### ISSUE TITLE
Startup Default - Move View>Edit>"Lock Object Mode" to viewport Options dropdown

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#3748]
![image](https://github.com/Bforartists/Bforartists/assets/35357409/b4424193-e674-4a2d-89c4-f770ff4c72dd)

Since rigging with this off is painful and being able to quickly switch modes with the number row... it's not really an issue.. Not sure when this got turned back on or if we ever turned this off..

**Comment by ReinerBforartists** [comment]
Mh, i don't think we should turn it off. This is something more advanced, isn't required for average work, and would bring new users into trouble. To turn it off brings more freedom for sure. But more freedom also introduces more confusion. One wrong click ...

Unfortunately i can't even find it in Blender to check the state there. Where did they hide it?

**Comment by Draise14** [comment]
3d view - Property shelf - View tab - Edit panel.

For weight painting UX, you literally cannot weight Paint till it is off. It would be a broken UX. 

When you have lock on you have to manually switch mode out from what you paint, select armature, switch mode in, choose bone, select mode out, select weight object, select weight mode in, paint - every single bone selection change to paint...... or use the list and guess what name what bone has... 

Without it, you can directly select bones while remaining in paint mode with a single click: armature to pose mode, select both, select active to weight paint object, weight Paint mode, select bone, paint ...

**Comment by Draise14** [comment]
Ah. In Blender it is in the header menus under Edit for some reason.

**Comment by ReinerBforartists** [comment]
Thanks. Ah now i see. I haven't done any weightpainting for years. But it remains dangerous for everything else.

Hmm, but are you sure that you really need to switch out? I seem to remember that there is a hotkey for exactly that since 2.80 days when they made weightpainting worse.

**Comment by ReinerBforartists** [comment]
Edit, when you enter weight painting with both selected, then you can switch bones by holding down ctrl. No need to switch modes.

First select armature, hold down shift, then select mesh. And then you can switch bones in weight paint mode.

**Comment by Draise14** [comment]
**Another use case:**
You have two skins to the same armature, ei through clothes, accessories, or multiple body parts. When it's off, you can switch as expected with CTRL+LMB. When it is on, you can _ONLY_ switch to other meshes in the correct mode - ei... 1. You have to make sure all your selections will be in the right mode. 2. It will be impossible to select something not in the right mode to get it into the right mode. 

Here is an example: 
1. First is with it locked, then when it's on. You can't select the object (second skin) because it's not in weight paint mode. 
2. Second part: you can then select between both skins on the same armature with ease, even if you're not all in the correct mode, this allows to move things or add to the selection before switching it to weight paint. With the lock, you can't. You have to intentionally switch out, select the additional mesh, then switch into weight paint again. 
https://github.com/Bforartists/Bforartists/assets/35357409/a461205e-3ef6-40fc-9085-79acf75e4a8d

I still think for weight painting UX, this is inconvenient weight paint UX with first use. This lock is mainly only for intentional weight painting locking when you're on a very dense collections of complex characters, ei.. you don't wanna accidentally click on a shirt mesh - it's an opt-in because of the scope of the character. For multi-skin weight painting, you'd have to pre-know to unlock to weight paint for the expected UX. There is no user warning for an expected behaviour that you are locked - thus not being able to switch meshes in weight paint mode. The chances for a miss click is irrelevant due to user confirmation by holding down CTRL. The misclick to intentionally select something when holding down CTRL I think is.. more likely to frustrate.  

**Another Use case:**
You are working with vertex painting, while doing layout to assign scattering vertex group on a ground mesh. With it on, you would have to toggle out of a mode, select, toggle into the mode, paint and repeat inverse to go back to rock pillar placement. But with it off you hold down CTRL+LMB and you can immediately select the mesh you'd need in the right mode it's in (the ground plane)
![PowerToys_oDfZcMh5Qb](https://github.com/Bforartists/Bforartists/assets/35357409/4db8b2e7-6a9e-42fa-a301-e8088f00a2ef)

With Vertex Paint, you can toggle selection with CTLR+LMB, but it's useless if you're locked. A useless expected hotkey if you have multiple modes on various meshes, and no warning that you're in locked mode when trying to use that expected hotkey.. 

**Another Use Case:**
If we clear this lock up, this also allows a consistent workflow we could add to all the modes - opt-in CTRL+LMB select regardless of mode beaurocracy.
Example: I am in texture paint mode, I use CTRL+LMB click to select another object in vertex paint mode, I then use CTRL+LMB to select the texture paint mode object again, or I select CTRL+LMB to select another texture paint mode. No Need to swap in then out of modes to change selection. To remedy the hotkey map to do this, the old inverse paint with CTRL+LMB press is now click drag, and - they don't conflict. 

https://github.com/Bforartists/Bforartists/assets/35357409/e6a53209-8958-419c-a5d5-5a9a491c7ebc

This also would allow consistent selection CTRL+LMB even in sculpt mode to another object in texture paint mode without needing to swap out, then in again, then swap out, then in again over and over like a mad man. 
![PowerToys_ZLX4BLDWnm](https://github.com/Bforartists/Bforartists/assets/35357409/29993942-9ed6-4230-93a6-b937061f341f)

**Comment by ReinerBforartists** [comment]
You need to have them all selected, that's all. At least from how i know it. I stay with my opinion, to make this default does more harm than good.,

**Comment by Draise14** [comment]
What is the harm?

**Comment by ReinerBforartists** [comment]
You may have more in the scene than just what you edit. And then one wrong click can select what you don't want to edit.  And this feature doesn't only affect rigging. But every edit mode. That's the danger here, and that's why it is optional and not default in Blender.

The way you work is very advanced. You need the advanced features. But the average user may not even rig a mesh with more than one meshpart at once. And that's what we need to aim at, the average user.

I feel your pain, so what about a compromise? We could add this menu feature to a more prominent location. It is an option, so we could move it into the options for example. Saves one click and digging. And is faster accessible for you then. Or add it to the Alighn View Buttons addon or the tool shelf, so that you can access it quickly top level UI.

Options would a bit more complicated, since there is more than one options menu. Bascially every object type has its own options menu in edit mode. Or no options menu at all.

![options1](https://github.com/Bforartists/Bforartists/assets/13919702/18db82d1-322b-4d8d-98bd-26e9d3314679)

![greasepencil](https://github.com/Bforartists/Bforartists/assets/13919702/932c53fa-61cb-4523-915d-4da4a293b2fe)

**Comment by ReinerBforartists** [comment]
Here a not to overlook solution ^^

![nottooverlook](https://github.com/Bforartists/Bforartists/assets/13919702/db468011-6961-44fa-b6c6-2a556f97ab84)

**Comment by Draise14** [comment]
Location is relatively fine - this usually is not toggled, only once as a user preference. Ideal location would be a user preference for all editors but that would be for Blender. 

I'll test if this is a global settig or if it is per editor then close. I'll do the other consistent CTRL+Lmb select then expose the operator to header in another task.

**Comment by Draise14** [comment]
Indeed it is a global setting for all 3D Viewports, so it's more a user preference for the Viewport tab.

**Comment by Draise14** [comment]
Updated task. I'm ok to default off, but I think this may help users know it's a preference, not a viewport specific view thing.

**Comment by Draise14** [comment]
![image](https://github.com/Bforartists/Bforartists/assets/35357409/c9e854df-800a-4dce-8692-43cf246fda8f)

You toggle it at the preferences toggle in blender from header.

**Comment by ReinerBforartists** [comment]
Yes i know now. I know it would fit into the preferences too, but i would keep it in reach like in Blender.

**Comment by Draise14** [comment]
![image](https://github.com/Bforartists/Bforartists/assets/35357409/bd392220-063f-4843-9844-9d90e720a557)

So here would make more sense? It's a 3D View option for selection, under the options panel? It would be one click less if you're using the property shelf.

**Comment by ReinerBforartists** [comment]
Yes, this would imho be the best solution. But you might want to add more options menus dropdowns then. Metaballs doesn't have an options menu in edit mode.

**Comment by Draise14** [comment]
Went through... 

Options Drop down needs to be added here to these:
- Text
- Metaballs
- Surface
- Curve
- Lattice

**Comment by ReinerBforartists** [comment]
> Grease Pencil
> 
> * Sculpt Mode
> * Vertex Paint Mode
> * Weight Paint Mode
> * Vertex Paint Mode

Isn't it exclusively to the edit mode?

**Comment by Draise14** [comment]
Nope. You can toggle selection with the same operator as weight paint mode if key exists. It exists in mesh vertex paint mode for some reason.

**Comment by ReinerBforartists** [comment]
Oki. Then it might be a better idea to create a general options dropdown that shows for all objects and in all modes. And merge the currently cluttered panels into this one panel.

**Comment by ReinerBforartists** [comment]
Edit, this would also allow us finally to move the rest of the options from the View menu into the options panel. But this would be another task then.

![viewoptions](https://github.com/Bforartists/Bforartists/assets/13919702/00c536bf-aecf-49df-9ba7-9f2a57f36292)

**Comment by ReinerBforartists** [comment]
Interestingly the toggle is now on by default.

![onbydefault](https://github.com/Bforartists/Bforartists/assets/13919702/201ed432-d6c2-4243-84c1-be74e345aaa7)

**Comment by Draise14** [comment]
The more I think about this in the options panel, it makes more sense.

**Comment by Draise14** [comment]
![bforartists_H8suBVeySe](https://github.com/Bforartists/Bforartists/assets/35357409/1db72dda-8a45-485c-b2e5-5afa208eff7f)

Like this?

**Comment by Draise14** [comment]
> Edit, this would also allow us finally to move the rest of the options from the View menu into the options panel. But this would be another task then.
> 
> ![viewoptions](https://private-user-images.githubusercontent.com/13919702/266769268-00c536bf-aecf-49df-9ba7-9f2a57f36292.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MzYwNDcsIm5iZiI6MTcxMTczNTc0NywicGF0aCI6Ii8xMzkxOTcwMi8yNjY3NjkyNjgtMDBjNTM2YmYtYWVjZi00OWRmLTliYTctOWYyYTU3ZjM2MjkyLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI5VDE4MDkwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmY2EyOGE0ODJjNGY5YzNmZDIyMmJlN2Y4NGQxZjNmZGZiMDAxYzczOGI2Zjc2ZTQ0ODI2NWEzNzRjMDM3ZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.agxrIT1W4j9K4QY-GUeY--JiqSdSyIB9LsosDN5nUvA)

Thinking about it, a global "view options" drop down is what you were talking about? That might make more sense..

**Comment by Draise14** [comment]
The more I think about it, the more I see it as a breaking inconsistent change. We have global properties for things either in header options panels. The 3D View has options.... per tool/mode. 

This is global. So the 3D View > View > Edit panel is.. in fact, the most logical discoverable way to do it, and to code it in that way is easiest, unless we make a completely new drop down somewhere for that one prop, or put it into a weird unrecoverable area (like the shader drop down).
